lib/src/generator/generate_header.dart

```diff
@@ -215,11 +216,28 @@ void _generateEnum({
   buffer.writeln(
       '/// - if (LocaleSettings.currentLocale == $baseLocaleEnumConstant) // locale check');
 
-  buffer.writeln('enum $enumName {');
+  buffer.writeln('class $enumName extends BaseAppLocale {');
```

Comment:
Do we still have exhausted `switch` calls?

Reply:
See the new code